### PR TITLE
Weapon Categories: Eskrima

### PR DESCRIPTION
**Pull request by Termineitor244**
#### Summary
Balance "Add weapon categories for Eskrima"

#### Purpose of change

Partially implements #51867.
Eskrima: Most of its categories are already implemented, just needing a little review and adding improvised weapons as a category for the hammer and other every day items already used by the style.

#### Describe the solution

| Weapon Category  | Martial Art | Includes |
| ------------- | ------------- | ------------- |
| KNIVES | Eskrima, Ninjutsu, Krav Maga  | All kind of knives |
| BATONS| Eskrima, Krav Maga  | Batons, tonfas an weapons of similar shape |
| BLADED_FARMING | Eskrima, Ninjutsu  | Bladed farming tools and the machetes |
| CLAWS | Eskrima, Ninjutsu  | Bionic and Tiger claws |
| SHIVS  | Eskrima  | Tiny, stabbing weapons |

KNIVES, BATONS, BLADED_FARMING and CLAWS are already used by Krav Maga or Ninjutsu, the only additional knife added in this PR was the Primitive Knife, and 3 new batons were added too, two kinds of pipes and the walking cane (All of these already in use by Eskrima, but lacking in the others list), some other weapons were changed to batons too.

1 new weapon category:
- SHIVS
Eskrima users are trained in the use of improvised weapons many of these fall in what could be called a "Shiv", so the appropriate weapons were added to this category, and the rest (plank, bowling pin, etc.) were removed from use in the style (SHIV category and some batons changes are thanks to @kevingranade, more info in the comments).

I removed the pool cue because it was supposed to have been removed in the martial arts rebalance (#33173) because of its size.

#### Describe alternatives you've considered

-Another name for IMPROVISED_WEAPONS, adding this category to Krav Maga too, or maybe just delete the whole weapons from use in a martial art.
Edit: Improvised weapons removed when they couldn't be accommodated in existing categories (+SHIVS).

#### Testing

None.

#### Additional context

**There needs to be a way to display the categories in the martial art info menu.**

### CODE REVIEW

**Comment by Rabidweasel** [comment]
I'm not sure that copper pipe should be included in batons, it should either be improvised weapons or not included at all; it's not really well suited to be used as a weapon.

The steel pipe probably resembles an actual weapon enough that including it in batons makes sense.

**Comment by Termineitor244** [comment]
They have the same length (60 cm), so I thought both could be justified as improvised batons, even if they are bad choices as weapons.

**Comment by kevingranade** [comment]
The question isn't whether it's a good weapon, the question is whether it's similar enough to the weapons used by the martial art that you can execute it's techniques.

File: data/json/items/melee/bludgeons.json [hunk 1]
```diff
@@ -287,6 +287,7 @@
     "material": [ "wood" ],
     "techniques": [ "WBLOCK_1" ],
     "flags": [ "DURABLE_MELEE" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ],
```
Comment (kevingranade) [inline data/json/items/melee/bludgeons.json:290]: Let's walk this one back, we don't need a special annotation for goofy weapons.
```suggestion
```

File: data/json/items/resources/wood.json [hunk 2]
```diff
@@ -107,7 +108,8 @@
     "bashing": 10,
     "price": 1000,
     "price_postapoc": 10,
-    "flags": [ "FIREWOOD" ]
+    "flags": [ "FIREWOOD" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/resources/wood.json:112]:
```suggestion
    "flags": [ "FIREWOOD" ]
```

File: data/json/items/melee/bludgeons.json [hunk 3]
```diff
@@ -400,6 +402,7 @@
     "longest_side": "20 cm",
     "cutting": 6,
     "flags": [ "SHEATH_KNIFE", "CONDUCTIVE", "FRAGILE_MELEE" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ],
```
Comment (kevingranade) [inline data/json/items/melee/bludgeons.json:405]:
```suggestion
    "weapon_category": [ "SHIV" ],
```

File: data/json/items/melee/bludgeons.json [hunk 4]
```diff
@@ -964,7 +967,8 @@
     "bashing": 1,
     "cutting": 3,
     "to_hit": -3,
-    "flags": [ "STAB", "SHEATH_KNIFE", "CONDUCTIVE" ]
+    "flags": [ "STAB", "SHEATH_KNIFE", "CONDUCTIVE" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/melee/bludgeons.json:971]:
```suggestion
    "weapon_category": [ "SHIV" ]
```

File: data/json/items/resources/misc.json [hunk 5]
```diff
@@ -12,6 +12,7 @@
     "volume": "250 ml",
     "cutting": 10,
     "flags": [ "SPEAR" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ],
```
Comment (kevingranade) [inline data/json/items/resources/misc.json:15]:
```suggestion
    "weapon_category": [ "SHIV" ],
```

File: data/json/items/resources/misc.json [hunk 6]
```diff
@@ -61,6 +62,7 @@
     "volume": "250 ml",
     "cutting": 8,
     "flags": [ "SPEAR", "SHEATH_KNIFE" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ],
```
Comment (kevingranade) [inline data/json/items/resources/misc.json:65]:
```suggestion
    "weapon_category": [ "SHIV" ],
```

File: data/json/items/resources/misc.json [hunk 7]
```diff
@@ -248,6 +250,7 @@
...
```
Comment (kevingranade) [inline data/json/items/resources/misc.json:253]:
```suggestion
    "weapon_category": [ "SHIV" ],
```

File: data/json/items/resources/wood.json [hunk 8]
```diff
@@ -44,6 +44,7 @@
     "material": [ "wood" ],
     "techniques": [ "WBLOCK_1" ],
     "flags": [ "TRADER_AVOID", "FIREWOOD" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ],
```
Comment (kevingranade) [inline data/json/items/resources/wood.json:47]:
```suggestion
    "weapon_category": [ "BATON" ],
```

File: data/json/items/tool/misc.json [hunk 9]
```diff
@@ -780,7 +780,8 @@
     "symbol": "/",
     "color": "magenta",
     "techniques": [ "WBLOCK_1" ],
-    "flags": [ "STAB", "RAIN_PROTECT", "SHEATH_SWORD" ]
+    "flags": [ "STAB", "RAIN_PROTECT", "SHEATH_SWORD" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/tool/misc.json:784]:
```suggestion
    "flags": [ "STAB", "RAIN_PROTECT", "SHEATH_SWORD", "FRAGILE_MELEE" ]

```

File: data/json/items/tool/stationary.json [hunk 10]
```diff
@@ -63,7 +63,8 @@
     "symbol": ",",
     "color": "light_red",
     "qualities": [ [ "CUT", 1 ], [ "BUTCHER", 3 ] ],
-    "flags": [ "STAB", "BELT_CLIP", "SHEATH_KNIFE" ]
+    "flags": [ "STAB", "BELT_CLIP", "SHEATH_KNIFE" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/tool/stationary.json:67]:
```suggestion
    "weapon_category": [ "SHIV" ]
```

File: data/json/items/tool/workshop.json [hunk 11]
```diff
@@ -368,7 +368,8 @@
     "color": "brown",
     "qualities": [ [ "HAMMER", 3 ], [ "PRY", 1 ], [ "HAMMER_FINE", 1 ], [ "PRYING_NAIL", 1 ] ],
     "use_action": [ "HAMMER", "CROWBAR" ],
-    "flags": [ "BELT_CLIP", "NONCONDUCTIVE" ]
+    "flags": [ "BELT_CLIP", "NONCONDUCTIVE" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/tool/workshop.json:372]:
```suggestion
    "weapon_category": [ "BATON" ]
```

File: data/json/items/tool/workshop.json [hunk 12]
```diff
@@ -810,6 +811,7 @@
     "symbol": ";",
     "color": "brown",
     "qualities": [ [ "HAMMER", 2 ] ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ],
```
Comment (kevingranade) [inline data/json/items/tool/workshop.json:814]:
```suggestion
```

File: data/json/items/tool/workshop.json [hunk 13]
```diff
@@ -830,7 +832,8 @@
     "symbol": ";",
     "color": "yellow",
     "qualities": [ [ "SCREW", 1 ] ],
-    "flags": [ "SPEAR", "BELT_CLIP" ]
+    "flags": [ "SPEAR", "BELT_CLIP" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/tool/workshop.json:836]:
```suggestion
    "weapon_category": [ "SHIV" ]
```

File: data/json/items/tool/workshop.json [hunk 14]
```diff
@@ -1189,7 +1192,8 @@
     "symbol": ";",
     "color": "light_gray",
     "qualities": [ [ "WRENCH", 2 ], [ "WRENCH_FINE", 1 ], [ "WHEEL_FAST", 1 ] ],
-    "flags": [ "BELT_CLIP" ]
+    "flags": [ "BELT_CLIP" ],
+    "weapon_category": [ "IMPROVISED_WEAPONS" ]
```
Comment (kevingranade) [inline data/json/items/tool/workshop.json:1196]:
```suggestion
    "weapon_category": [ "BATON" ]
```

**Comment by kevingranade** [comment]
Regardless of whether a style says they are trained with "improvised weapons", the reality is they train with classes of weapons, and possibly practice using unusual variations.
If we start throwing weapons into undifferentiated groups, then the act of classifying them becomes pointless.
In this case all of the improvised weapons fall into either "baton" or "tiny stabbing weapons", which I've called "shivs".
I went ahead and removed some that were evidently thrown in with minimal justification, a 2x4 is not a baton, you can't grip it like one, same with a bowling pin, which is perversely too big and heavy to use as a baton.

At the same time, this widens the scope of what some other baton styles can use, which is fine.

File: data/json/martialarts.json [hunk 15]
```diff
@@ -310,60 +310,7 @@
...
-      "tanto",
-      "tanto_inferior",
-      "tanto_fake",
-      "tonfa",
-      "tonfa_wood",
-      "umbrella",
-      "wasp_sting",
-      "wrench"
-    ]
+    "weapon_category": [ "KNIVES", "BATONS", "BLADED_FARMING", "CLAWS", "IMPROVISED_WEAPONS" ]
```
Comment (Termineitor244) [inline data/json/martialarts.json:313]:
```suggestion
    "weapon_category": [ "KNIVES", "BATONS", "BLADED_FARMING", "CLAWS" ]
```

**Comment by Termineitor244** [comment]
> 
> 
> Regardless of whether a style says they are trained with "improvised weapons", the reality is they train with classes of weapons, and possibly practice using unusual variations. If we start throwing weapons into undifferentiated groups, then the act of classifying them becomes pointless. In this case all of the improvised weapons fall into either "baton" or "tiny stabbing weapons", which I've called "shivs". I went ahead and removed some that were evidently thrown in with minimal justification, a 2x4 is not a baton, you can't grip it like one, same with a bowling pin, which is perversely too big and heavy to use as a baton.
> 
> At the same time, this widens the scope of what some other baton styles can use, which is fine.

Thank you! This gives more direction to the weapon categories, the only remaining issue could be with some mythical martial arts... I will have to talk with @Hymore246 about it.

I went and removed the category from the martial art, I'm not sure why a hammer would be a baton but an improvised hammer would not, but whatever, the PR should be ready with this.

**Comment by Termineitor244** [comment]
I changed "SHIV" to "SHIVS" to keep consistency with the rest of the weapon categories, and the new batons were appropriately changed from "BATON" to "BATONS" since the is not a "BATON" category.

An alternative would be change them all to singular form, but I would have to change the already implemented weapons of krav maga and ninjutsu too, if that is desired then it would be best to know before changing more weapons as I continue with the rest of the styles.

**Comment by kevingranade** [comment]
> I changed "SHIV" to "SHIVS" to keep consistency with the rest of the weapon categories, and the new batons were appropriately changed from "BATON" to "BATONS" since there is not a "BATON" category.

Sorry about that, that's what I get for making edits on the fly, following your plural naming scheme is fine.

File: data/json/items/tool/misc.json [hunk 16]
```diff
@@ -780,7 +780,8 @@
     "symbol": "/",
     "color": "magenta",
     "techniques": [ "WBLOCK_1" ],
-    "flags": [ "STAB", "RAIN_PROTECT", "SHEATH_SWORD" ]
+    "flags": [ "STAB", "RAIN_PROTECT", "SHEATH_SWORD", "FRAGILE_MELEE" ]
+
```
Comment (Termineitor244) [inline data/json/items/tool/misc.json:784]:
```suggestion
```

**Comment by Termineitor244** [comment]
> 
> 
> Sorry about that, that's what I get for making edits on the fly, following your plural naming scheme is fine.

Don't worry! It's something that happens sometimes haha

File: data/json/items/tool/misc.json (hunk 16 above)
Comment (kevingranade) [inline data/json/items/tool/misc.json]:
```suggestion
```

**Pull request merged**